jwst/ami/oifits.py

```diff
@@ -0,0 +1,763 @@
+            self.method = 'median'
+        # set these as attributes (some may exist and be overwritten)
+        if self.method == 'multi':
+            self.vis2 = self.fa2.T
+            self.e_vis2 = np.zeros(self.vis2.shape)
+            self.visamp = self.fa.T
+            self.e_visamp = np.zeros(self.visamp.shape)
+            self.visphi = self.fp.T
+            self.e_visphi = np.zeros(self.visphi.shape)
+            self.cp = self.cp.T
+            self.e_cp = np.zeros(self.cp.shape)
+            self.camp = self.ca.T
+            self.e_camp = np.zeros(self.camp.shape)
+            self.pist = self.pistons.T
+            self.e_pist = np.zeros(self.pist.shape)
+
+        # apply sigma-clipping to uncertainties
+        # sigma_clipped_stats returns mean, median, stddev. nsigma=3, niters=5
+        elif self.method == 'median':
+            _, self.vis2, self.e_vis2 = sigma_clipped_stats(self.fa2, axis=0)
+            _, self.visamp, self.e_visamp = sigma_clipped_stats(self.fa, axis=0)
+            _, self.visphi, self.e_visphi = sigma_clipped_stats(self.fp, axis=0)
+            _, self.cp, self.e_cp = sigma_clipped_stats(self.cp, axis=0)
+            _, self.camp, self.e_camp = sigma_clipped_stats(self.ca, axis=0)
+            _, self.pist, self.e_pist = sigma_clipped_stats(self.pistons, axis=0)
+
+        else: # take the mean
+            self.vis2, _, self.e_vis2 = sigma_clipped_stats(self.fa2, axis=0)
+            self.visamp, _, self.e_visamp = sigma_clipped_stats(self.fa, axis=0)
+            self.visphi, _, self.e_visphi = sigma_clipped_stats(self.fp, axis=0)
+            self.cp, _, self.e_cp = sigma_clipped_stats(self.cp, axis=0)
+            self.camp, _, self.e_camp = sigma_clipped_stats(self.ca, axis=0)
+            self.pist, _, self.e_pist = sigma_clipped_stats(self.pistons, axis=0)
+
+        # prepare arrays for OI_ARRAY ext
+        self.staxy = info4oif.ctrs_inst
+        N_ap = len(self.staxy)
+        tel_name = ['A%i' % x for x in np.arange(N_ap)+1]
+        sta_name = tel_name
+        diameter = [0] * N_ap
+
+        staxyz = []
+        for x in self.staxy:
+            a = list(x)
+            line = [a[0], a[1], 0]
+            staxyz.append(line)
+
+        sta_index = np.arange(N_ap) + 1
+
+        pscale = info4oif.pscale_mas/1000.  # arcsec
+        isz = self.ff.scidata.shape[1]  # Size of the image to extract NRM data
+        fov = [pscale * isz] * N_ap
+        fovtype = ['RADIUS'] * N_ap
+
+        # prepare info for OI_TARGET ext
+        name_star = info4oif.objname
+
+        customSimbad = Simbad()
+        customSimbad.add_votable_fields('propermotions', 'sptype', 'parallax')
+
+        # Add informations from Simbad:
+        if name_star == 'UNKNOWN':
+            ra, dec, spectyp = [0], [0], ['unknown']
+            pmra, pmdec, plx = [0], [0], [0]
+        else:
+            try:
+                query = customSimbad.query_object(name_star)
```

Comment:
Can some (or all) of this be read from the existing meta data (or defaults/nans filled) instead of making a Simbad query? This would allow dropping the astroquery dependency and I'm not sure making queries to external services will work well in operations?

Reply:
It probably can be replaced; I'll look into this.